lib/galaxy/managers/collections.py

```diff
@@ -296,7 +296,9 @@ def copy(self, trans, parent, source, encoded_source_id, copy_elements=False):
         source_hdca = self.__get_history_collection_instance(trans, encoded_source_id)
         copy_kwds = {}
         if copy_elements:
-            copy_kwds["element_destination"] = parent
+            copy_kwds["element_destination"] = parent  # e.g. a history
+        if copy_dataset_instance_attributes is not None:
+            copy_kwds["copy_dataset_instance_attributes"] = copy_dataset_instance_attributes
```

Comment:
```suggestion
            copy_kwds["element_destination"] = parent  # e.g. a history
            copy_kwds["dataset_instance_attributes"] = dataset_instance_attributes
```
Since we correctly check that we only change dataset attributes when we copy the collection in the API we can use a single if here. If `dataset_instance_attributes` is None we replace it with an empty dict in the copy method, so passing in None is not a problem here.